src/core/distmatrix.jl

```diff
@@ -100,7 +100,7 @@ for (elty, ext) in ((:ElInt, :i),
             return rv[]
         end
 
-        function processPullQueue(A::DistMatrix{$elty}, buf::Array{$elty,2})
+        function processPullQueue(A::DistMatrix{$elty}, buf::AbstractArray{$elty,2})
```

Comment:
I don't think this is safe. The `ccall` creates a pointer to `buf` so it has to be a contiguous memory block.